gazelle_haskell_modules/rule_generation.go

```diff
@@ -265,21 +272,35 @@ func concatRuleInfos(xs [][]*RuleInfo) []*RuleInfo {
 	return ys
 }
 
-// Collects the source files referenced in the given expression
-func srcsFromRule(pkgRoot string, expr build.Expr) ([]string, error) {
```

Comment:
Ah well, it isn't actually thrown away - I just renamed it to `getSrcs`, and what you described is exactly what `getSrcs` does. I'll factor out the two parts into two smaller functions since they're not really related at all.